call-profile/main.go

```diff
@@ -0,0 +1,566 @@
+		Phone:       diff.Phone,
+		YOE:         diff.YOE,
+		Company:     diff.Company,
+		Designation: diff.Designation,
+		GithubId:    diff.GithubId,
+		LinkedIn:    diff.LinkedIn,
+		TwitterId:   diff.TwitterId,
+		InstagramId: diff.InstagramId,
+		Website:     diff.Website,
+	}
+}
+
+func resToDiff(res Res, userId string) Diff {
+	return Diff{
+		UserId:      userId,
+		Timestamp:   time.Now(),
+		Approval:    "PENDING",
+		FirstName:   res.FirstName,
+		LastName:    res.LastName,
+		Email:       res.Email,
+		Phone:       res.Phone,
+		YOE:         res.YOE,
+		Company:     res.Company,
+		Designation: res.Designation,
+		GithubId:    res.GithubId,
+		LinkedIn:    res.LinkedIn,
+		TwitterId:   res.TwitterId,
+		InstagramId: res.InstagramId,
+		Website:     res.Website,
+	}
+}
+
+func diffToMap(diff Diff) map[string]interface{} {
+	return map[string]interface{}{
+		"userId":       diff.UserId,
+		"timestamp":    diff.Timestamp,
+		"approval":     diff.Approval,
+		"first_name":   diff.FirstName,
+		"last_name":    diff.LastName,
+		"email":        diff.Email,
+		"phone":        diff.Phone,
+		"yoe":          diff.YOE,
+		"company":      diff.Company,
+		"designation":  diff.Designation,
+		"github_id":    diff.GithubId,
+		"linkedin_id":  diff.LinkedIn,
+		"twitter_id":   diff.TwitterId,
+		"instagram_id": diff.InstagramId,
+		"website":      diff.Website,
+	}
+}
+
+/*
+ Setting Constants Map
+*/
+var Constants map[string]string = map[string]string{
+	"ENV_DEVELOPMENT":         "DEVELOPMENT",
+	"ENV_PRODUCTION":          "PRODUCTION",
+	"STORED":                  "stored",
+	"FIRE_STORE_CRED":         "firestoreCred",
+	"PROFILE_SERVICE_HEALTH":  "PROFILE_SERVICE_HEALTH",
+	"PROFILE_SKIPPED":         "PROFILE_SKIPPED",
+	"PROFILE_DIFF_STORED":     "PROFILE_DIFF_STORED",
+	"STATUS_BLOCKED":          "BLOCKED",
+	"PROFILE_SERVICE_BLOCKED": "PROFILE_SERVICE_BLOCKED",
+	"NOT_APPROVED":            "NOT APPROVED",
+	"PROFILE_SKIPPED_DUE_TO_UNAUTHENTICATED_ACCESS_TO_PROFILE_DATA": "profileSkippedDueToUnAuthenticatedAccessToProfileData",
+	"PROFILE_SKIPPED_DUE_TO_ERROR_IN_GETTING_PROFILE_DATA":          "profileSkippedDueToErrorInGettingProfileData",
+	"SKIPPED_SAME_LAST_REJECTED_DIFF":                               "skippedSameLastRejectedDiff",
+	"SKIPPED_SAME_LAST_PENDING_DIFF":                                "skippedSameLastPendingDiff",
+	"SKIPPED_CURRENT_USER_DATA_SAME_AS_DIFF":                        "skippedCurrentUserDataSameAsDiff",
+	"SKIPPED_OTHER_ERROR":                                           "skippedOtherError",
+	"SKIPPED_VALIDATION_ERROR":                                      "validation error",
+}
+
+/*
+ Setting Firestore Key for development/production
+*/
+func getFirestoreKey() string {
+	if os.Getenv(("environment")) == Constants["ENV_DEVELOPMENT"] {
+		return os.Getenv(Constants["FIRE_STORE_CRED"])
+	} else if os.Getenv(("environment")) == Constants["ENV_PRODUCTION"] {
+		var parameterName string = Constants["FIRE_STORE_CRED"]
```

Comment:
parameterName is declared as a string, but the value that is assigned to it is a constant. This means that the value of parameterName will always be the same, regardless of the environment that the function is running in.

Reply:
Yes because in develop environment, we stored it as the environment variable with the constant parameter name. But in production, we stored it in the parameter manager in the AWS system manager. So methods of accessing are different.